### PR TITLE
Api docs for permissoned token

### PR DESCRIPTION
**Pull request by ratik21**

### CODE REVIEW

**Review by robert-zaremba — COMMENTED** [review]
I think it's better to reformat this API to more low level description:

* use same sections (common, admin...)
* clarify the notation - in smart contracts we don't have functions, we have transactions and transaction attributes.
* describe where this use-case is implemented in `scripts`.

File: examples/permissioned-token/README.md [hunk 1]
```diff
@@ -96,3 +96,49 @@ Developer hour links:
...
+
+### API Doc
+
+#### Common
+
+1. *totalSupply(assetIndex)*: returns total supply of the asset (*asset.total* - *asaReserveHolding.amount*).
+
+#### Admin
+1. *issue(address, amount)*: Group of 2 transactions
+   - *tx1*: Call to controller smart contract with a) application arg `str:issue` b) foreign Asset: `assetIndex` - signed by asset reserve (issuer).
```
[the remark this answers is not in the document]
Comment (ratik21) [reply examples/permissioned-token/README.md:108]: > Call to controller stateful smart contract

This is what we have now. Let's keep it that way (as we do `CallNoOpSSC` in transaction type, which indicates calling SSC).

**Review by man-of-code — APPROVED** [review]
LGTM

File: examples/permissioned-token/README.md [hunk 2]
```diff
@@ -96,3 +96,50 @@ Developer hour links:
 - *implementation of the solution above using algob*: [examples/restricted-assets](/examples/restricted-assets)
 
 Link to spec of current template: [https://paper.dropbox.com/doc/Algob-Security-Token-Template-FR2LXhVg3edevYPBQZw6F](https://paper.dropbox.com/doc/Algob-Security-Token-Template-FR2LXhVg3edevYPBQZw6F)
+
+### API Doc
+Below we describe different use cases. We describe a use case using a function notation (along with transaction group structure). We implement that function in scripts, but for direct integration in smart contracts, you need to construct transactions as described below.
+
```
Comment (robert-zaremba) [inline examples/permissioned-token/README.md:102]:
```suggestion
### Use Case API

Below we describe different use cases implemented by the smart contract suite. We use a function notation for a use-case, along with description of transaction group which has to be created. We implemented that use cases using functions in `scripts` directory. For direct integration in smart contracts, you need to construct transactions as described below.

```

File: examples/permissioned-token/README.md [hunk 3]
```diff
@@ -96,3 +96,50 @@ Developer hour links:
...
+#### Common
```
Comment (robert-zaremba) [inline examples/permissioned-token/README.md:103]:
```suggestion
#### Query
```

File: examples/permissioned-token/README.md [hunk 4]
```diff
@@ -96,3 +96,50 @@ Developer hour links:
...
+
+1. [*totalSupply(assetIndex)*](examples/permissioned-token/scripts/common/common.js): returns total supply of the asset (*asset.total* - *asaReserveHolding.amount*).
```
Comment (robert-zaremba) [inline examples/permissioned-token/README.md:105]: Please add a line how to get user token holding (this is a standard ASA query).

File: examples/permissioned-token/README.md [hunk 5]
```diff
@@ -96,3 +96,50 @@ Developer hour links:
...
+
+2. [*kill()*](examples/permissioned-token/scripts/admin/kill.js): Application call to controller with a) appArg `str:kill` and b) foreignAsset `assetIndex`- signed by asset manager.
+
+3. [*forceTransfer (fromAddress, toAddress, amount)*](examples/permissioned-token/scripts/admin/force-transfer.js): Group of 4 transactions
+   - *tx1*: Call to controller smart contract with a) application arg `str:force_transfer` b) foreign Asset: `assetIndex` - signed by asset manager.
+   - *tx2*: Asset clawback transaction from `fromAddress` to `toAddress`, amount = `amount`.
+   - *tx3*: Payment transaction to clawback (contract account) with amount >= fee of *tx2*
+   - *tx4*: Call to permissions smart contract with a) application arg `str:transfer` b) app accounts: [`fromAddress`, `toAddress`].
+
+    **NOTE**: *tx3*, *tx4* can be signed by anyone but they should be present in group. If receiver of `forceTransfer` is the current asset reserve OR if *tx4* is asset config txn (to update reserve address), then it can bypass permission checks.
```
Comment (robert-zaremba) [inline examples/permissioned-token/README.md:120]:
```suggestion
    **NOTE**: *tx3*, *tx4* can be signed by anyone but they must be present in group (they validate conditions). The signer will pay transaction fees. If receiver of `forceTransfer` is the current asset reserve OR if *tx4* is asset config txn (to update reserve address), then a permissions smart contract call is not required.
```

Comment (robert-zaremba) [reply examples/permissioned-token/README.md:120]: Maybe we should reword it. In the second sentence of the note, a reader may have impression that we are still talking about `forceTransfer` - so why do we talk about asset config tx in this context? I think we should move that part to the use case below.

File: examples/permissioned-token/README.md [hunk 6]
```diff
@@ -96,3 +96,50 @@ Developer hour links:
...
+### User
+
+1. [*transfer (fromAccount, toAddress, amount)*](examples/permissioned-token/scripts/user/transfer.js): Group of 4 transactions
+   - *tx1*: Call to controller smart contract with application arg `str:transfer` - signed by *fromAccount*.
+   - *tx2*: Asset clawback transaction from `fromAccount.address` to `toAddress`, amount = `amount`.
+   - *tx3*: Payment transaction to clawback (contract account) with amount >= fee of *tx2*
+   - *tx4*: Call to permissions smart contract with a) application arg `str:transfer` b) app accounts: [`fromAccount.address`, `toAddress`].
+
+2.  [*optOut (account)*](examples/permissioned-token/scripts/user/opt-out.js): Asset transfer transaction from `account.address` to `account.address`, amount = 0, **closeRemainderTo** = asset.creator.
+    **NOTE**: User opts out of the asset to creator (transferring account's asset holdings to the creator's account), so if creator is not the asset reserve, then the tokens from which user opted-out will still be in circulation.
```
Comment (robert-zaremba) [inline examples/permissioned-token/README.md:139]:
```suggestion
    **NOTE**: User opts out from the permissioned token. Algorand will transfer all his tokens to the ASA creator not the ASA reserve account (by definition of Algorand opt-out transaction).
```

File: examples/permissioned-token/README.md [hunk 7]
```diff
@@ -96,3 +96,50 @@ Developer hour links:
...
+
+### Permissions
+
+1. [*whitelist (permissionsManager, address)*](examples/permissioned-token/scripts/permissions/whitelist.js): NoOp call to the permissions smart contract with a) application arg `str:add_whitelist` b) app accounts: [`address`] - signed by *permissionsManager*.
```
Comment (robert-zaremba) [inline examples/permissioned-token/README.md:143]:
```suggestion
1. [*whitelist (permissionsManager, address)*](examples/permissioned-token/scripts/permissions/whitelist.js): NoOp call to the permissions smart contract with a) application arg `str:add_whitelist` b) app accounts: [`address`] - must signed by *permissionsManager*.
```

**Review by robert-zaremba — APPROVED** [review]
left few comments. pre-approving

File: examples/permissioned-token/README.md (hunk 7 above)
Comment (ratik21) [reply examples/permissioned-token/README.md:143]: updpating to "must be signed by.."

File: examples/permissioned-token/README.md (hunk 4 above)
Comment (ratik21) [reply examples/permissioned-token/README.md:105]: ok

File: examples/permissioned-token/README.md (hunk 5 above)
Comment (ratik21) [reply examples/permissioned-token/README.md]: ok, makes sense

**Pull request merged**